### DISCUSSION TITLE
Why hack a kid's game?

### DISCUSSION THREAD

**Discussion by PatheticMustan** [Prodigy-Hacking/ProdigyMathGameHacking#16] [(A) Anything]
The reason I created this repository was because I just wanted to have fun. I also wanted to inform Prodigy about how easy it was to hack.

What we make aren't really "hacks". Nothing cutting edge. The reason we're able to make such a comprehensive list of hacks is because of the way Prodigy is built.

`PIXI.game.prodigy.player.data.gold` is an integer that stores the amount of gold we have.
To "hack" our gold, all we need to do is change it to a higher number. It's literally just:
```
PIXI.game.prodigy.player.data.gold = 666;
```
It's really simple. I thought it was strange that such a popular game would have little/no protection against these types of things.
Once again, I only made this repository for fun.
But now, there's a WHOLE lot of people using this repo.

![image](https://user-images.githubusercontent.com/35382971/75086809-4ab8a100-5506-11ea-9673-e1153d3ed9d8.png)

As of now, we have had almost 900 people look at this repository. While I think hacking games is pretty cool, and fun, it's only cool when you're doing it for ***educational*** purposes.

If you hack games just to feel good, that isn't cool. I wasn't trying to distribute hacks everywhere.
I'm still going to develop new hacks (because it's fun, and I really want an anti-cheat), but please don't go around bragging about being a hacker.

Remember, if you want something, just ask nicely. If we can do it, we'll try, because y'all seem like such a nice bunch!

**Comment by ArcerionDev** [comment]
Omg that's a lot of visitors!

**Comment by GyarGG** [comment]
very well worded. Thank you for doing this, and yeah that's a lot of viewers!

**Comment by PatheticMustan** [comment]
It seems like somebody make a youtube video on us already.    ,':-(

**Comment by ArcerionDev** [comment]
That was fast. Give us the link…

**Comment by PatheticMustan** [comment]
Well actually, I'm not totally sure there's a video about US, but according to Github Traffic, we're getting 322 visits, 46 unique visitors from Youtube alone.

**Comment by ArcerionDev** [comment]
Wow, that's a pretty good amount for a Github Repository like this

**Comment by PatheticMustan** [comment]
https://prodigy-math-game.fandom.com/f/p/3100920787239247988/r/3116291309175130550

I uh... made a post about hacking.
Really hope Prodigy makes an anticheat

**Comment by ArcerionDev** [comment]
I really doubt they will, because they would have to rebuild their
functions from the ground up.
Right?

**Comment by PatheticMustan** [comment]
Basically. It would be a lot more work on their end than on ours.

**Comment by ArcerionDev** [comment]
Exactly. So by figuring this out, we've basically put them in a big
dilemma...

**Comment by ArcerionDev** [comment]
Right?

**Comment by ghost** [comment]
I can send Prodigy these hacks directly. They might patch them.

**Comment by PatheticMustan** [comment]
I'd really like to prioritize documentation/explanations in the [wiki](https://github.com/PatheticMustan/ProdigyMathGameHacking/wiki). Would anybody like to help?

**Comment by ArcerionDev** [comment]
Sure, I can do that.

**Comment by ArcerionDev** [comment]
Maybe...

**Comment by PatheticMustan** [comment]
Guys, please stop posting links to this repository on the Prodigy wiki.

**Comment by ArcerionDev** [comment]
Ok.

**Comment by PatheticMustan** [comment]
After I'm no longer busy with school, I'm going to start working on a menu, so that it's easier to use/call/add hacks...

**Comment by ArcerionDev** [comment]
Ok, that sounds good!

**Comment by ArcerionDev** [comment]
We have 16 forks wtf

**Comment by Exoticc** [comment]
> We have 16 forks wtf

Wow that is a lot of forks

**Comment by ArcerionDev** [comment]
🍴

**Comment by zoobidoobi007** [comment]
I just noticed something in the hacks list, and it was the free membership hack. Isn't that illegal? I mean if everyone who wasn't a member, or everyone who uses prodigy does that hack, prodigy will lose money and go bankrupt. Isn't it kind of like stealing? I mean, you didn't pay for the membership, so how is it fair to Prodigy that you can still get it? Just saying.

**Comment by zoobidoobi007** [comment]
BTW same with the get all pets hack. To get Epic Monsters/Pets, you need to buy a toy and then use the code on the toy. If everyone starts using the Get all monsters hack, then Prodigy will lose toys sales. Maybe just remove the epics from that hack? IDK any coding, I just look at cool hacks for games.

**Comment by ArcerionDev** [comment]
First of all, epics were discontinued. Second of all, it's their own fault
for not making an anti-cheat.

**Comment by PatheticMustan** [comment]
(I've been trying to push them to make an anticheat)

**Comment by ghost** [comment]
First of all, YES, Epics were discontinued. Second, it's not their fault, it's all the Epics' fault they do not have strong spells. Third of f*cking all, Prodigy should have, not dozens, but tons of Epics for people to purchase and use, or they can just use the get all pets and buddies hack. and... (goes to his room and rages)

**Comment by JustTheTruth** [comment]
> Notice: We are NOT trying to ruin Prodigy Math Game.
> This may look like we're trying to ruin the game, but we're helping make it better. By finding exploits, we help the developers find that there is an exploit, and the developers fix it, overall helping the game. Anyone who claims that we are 'malicious' or 'ruining the game', are wrong. Please don't spread misinformation.

You are correct in that these are exploits, and it is Prodigy's fault for not protecting themselves. You are incorrect both in that all of these exploits are legal and in the fact that you are doing this to help Prodigy.

In terms of legality, theft is illegal, and determining a way to bypass payment is theft, regardless of Prodigy's lack of security. If a cash register is left open and the cashier's back is turned, it doesn't mean the money is up for grabs. Theft is theft.

In terms of helping Prodigy, a random **_public_** GitHub account does not help push them to make an anticheat. If you actually cared about the product (or the effects of your 'fun'), you would email them the exploits instead of spreading them publicly. So either you're a liar, willfully ignorant, or stupid. None of those shed the best light on you (and won't hold up legally either!).

In terms of not ruining the experience or the game, think about the effects that these exploits have on the public experience, such as the PvP leaderboards. You can use these exploits to completely box out other users during a PvP season, devaluing their hard work. In fact, if your train of logic is to distribute and use these exploits in-game, hope Prodigy finds out about them because they are affecting the experience, and _then_ Prodigy will fix them, then you must admit that your exploits do have a negative impact on the experience.

TLDR: Your fun is not innocent, and your intentions are misplaced.

**Comment by PatheticMustan** [comment]
I personally have emailed them 3 times, and sent them a link to this very repository, multiple times.
Not only that, but "membership hacks" aren't bypassing ay payment screen. It only lasts until you reload the page, because all you do is change a boolean "isMember" from false to true. It's like calling inspect element "hacking". It's really not.

**Comment by JustTheTruth** [comment]
Inspect element would be the equivalent of purely looking at the Javascript and doing nothing with it - you can do very little damage with inspection. They are not one and the same. Just because it's easy to hack membership still doesn't make it legal or right. If I download a song/movie without paying, I'm not _necessarily_ bypassing a payment screen/method either. **_I'm accessing something that I would have had to pay for through conventional means_**. Hacking is not conventional in any shape or form when it comes to commerce, nor is it legal.

But I am not without both sympathy and empathy for you. I do recommend you brush up on your legal knowledge, as I do believe that _you_ believe what you are doing is legal. As a fellow programmer (and one who's been doing it for over 2 decades) I know what it's like to feel the kind of power that comes with discovering an exploit. Hell, as a gamer alone I have certainly found - and used - my share of exploits. But all within legal limits, because I know how easy it is to sometimes convince yourself that what you're doing isn't hurting anyone.

Tread carefully.

**Comment by PatheticMustan** [comment]
They patched so much stuff today! Yay!

**Comment by ghost** [comment]
I can't even complete shipwreck shore on the early account

**Comment by ghost** [comment]
> They patched so much stuff today! Yay!

What do you mean?

**Comment by ghost** [comment]
They put in an anti cheat today. If you open Inspect Element, it disconnects from the Prodigy server.

**Comment by ArcerionDev** [comment]
Yup

**Comment by ArcerionDev** [comment]
https://prodigy-cheat-dashboard.herokuapp.com/ Go here

**Comment by PatheticMustan** [comment]
It seems that every cheat has been patched. Clientside is very, very hard, and not worth the time. Serverside is also much harder now, not even sure if it's possible.

**Comment by MelnCat** [comment]
BeT

**Comment by Exoticc** [comment]
bEt

**Comment by ooo788** [comment]
beT

**Comment by ghost** [comment]
beT

**Comment by PatheticMustan** [comment]
bet

**Comment by ArcerionDev** [comment]
beet

**Comment by BoredFishRE** [comment]
my meet

**Comment by ghost** [comment]
lmao

**Comment by ooo788** [comment]
i actually wanted to know if u can reverse the bobby fancywoman thing?
cause on my main account it glitched and it just made myself look like him
:( and i had loads of stuff on my main so PLZZZZZZZ IM LITTERY DESPRATE

**Comment by PatheticMustan** [comment]
Unfortunately, Bobby Fancywoman cannot be reversed. It even gives you a warning! 🤨

**Comment by vibinyama** [comment]
actually using the menu its extraordinarily easy to reverse the effects of it

**Comment by Fe4rlessfan121** [comment]
PatheticMustan good work tho

**Comment by rusprice** [comment]
Going to close this so that it won't count as an actual issue (since it's just meta). You may still reply to this issue.

**Comment by TheArcadeKing** [comment]
I'll pay someone to PLEASE hep my son get gold. I'm willing to paypal, cashapp, venmo and i can provide his login. PLEEEEEEEEEEEASE SOMEONE HELP

**Comment by TheArcadeKing** [comment]
*help

**Comment by ghost** [comment]
> I'll pay someone to PLEASE hep my son get gold. I'm willing to paypal, cashapp, venmo and i can provide his login. PLEEEEEEEEEEEASE SOMEONE HELP

i'll do it for bitcoin. deal?

**Comment by ghost** [comment]
jkjk

**Comment by rusprice** [comment]
> I'll pay someone to PLEASE hep my son get gold. I'm willing to paypal, cashapp, venmo and i can provide his login. PLEEEEEEEEEEEASE SOMEONE HELP

Yeah I feel like this isn't his parent, sounds like it's him pretending. My senses also detect the author of this message of is <13

**Comment by PatheticMustan** [comment]
![image](https://user-images.githubusercontent.com/35382971/102515607-f31e8f80-405b-11eb-9611-01411f9160c1.png)
We've grown so much... :D

**Reply by PatheticMustan** [reply]
![image](https://user-images.githubusercontent.com/35382971/111052777-145d0500-842c-11eb-8112-d5dc69ceefde.png)
